src/lib/molecules/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,215 @@
+     * The Tooltip component size<br> Possible values: `default | small`
+     */
+    size?: 'default' | 'small';
+    /**
+     * Main content for the component.
+     */
+    text?: string;
+    /**
+     * Title for the component.
+     */
+    title?: string;
+    /**
+     * Style object, to have extra styles.
+     */
+    style?: CSSProperties;
+    /**
+     * The component will be visible without any action.
+     */
+    alwaysShow?: boolean;
+    /**
+     * Will display the component in the specified location.
+     */
+    customPosition?: ICustomPosition;
+    /**
+     * Any valid React node.
+     */
+    children: JSX.Element | JSX.Element[];
+    /**
+     * Disable/Enable auto repositions.
+     */
+    disableReposition?: boolean;
+    /**
+     * Positions where will be displayed the Tooltip relates the child component.<br> Possible values: `top | right | bottom | left`
+     */
+    position?: 'top' | 'right' | 'bottom' | 'left';
+    /**
+     * Tooltip padding from the target element
+     */
+    padding?: number;
+    /**
+     * Control with screenType  appearance of component
+     */
+    screenType?: 'desktop' | 'mobile';
+    /**
+     * In case of `false` value, the children component will rendered without Tooltip.
+     */
+    isVisible?: boolean;
+    /**
+     * The action will triggered when the Tooltip component will clicked.
+     */
+    onClick?: (e: MouseEvent) => void;
+}
+
+const FindAndMergeRef = <T extends { onClick: (e: PointerEvent, el: JSX.Element) => void }>(
+    children: JSX.Element | JSX.Element[],
+    childProps: T,
+    componentRef: (node: ReferenceType | null) => void
+) =>
+    Children.map(children, (el, i) => {
+        const newProps = {
+            ...childProps,
+            onClick: (e: PointerEvent) => childProps?.onClick(e, el),
+            ref: i === 0 ? componentRef : {}
+        };
+
+        if (el?.type === Fragment && el.props.children) {
+            return FindAndMergeRef(el.props.children, childProps, componentRef);
+        }
+        if (isForwardRef(el)) {
+            return FindAndMergeRef(el.type.render(el.props), newProps, componentRef);
+        }
+        if (typeof el?.type === 'string') {
+            return cloneElement(el, newProps);
+        }
+        if (typeof el?.type === 'function') {
+            return FindAndMergeRef(el.type(el.props), newProps, componentRef);
+        }
+
+        return cloneElement(el, newProps);
+    });
+
+const Tooltip: FC<ITooltipProps> = ({
+    children,
+    position = 'top',
+    size = 'default',
+    style,
+    text,
+    title,
+    customPosition,
+    alwaysShow,
+    disableReposition = false,
+    onClick = noop,
+    padding = 5,
+    screenType = 'desktop',
+    isVisible = true,
+    ...props
+}) => {
+    // @ts-ignore
+    const { geneUIProviderRef } = useContext(GeneUIDesignSystemContext);
+    const { isMobile } = useDeviceType(screenType);
+    const [isPopoverOpen, setPopoverState] = useState(false);
```

Comment:
I think you need to rename state variables to align the setter with the state name